src/machines/mlEphantManagerMachine2.ts

```diff
@@ -227,6 +227,15 @@ export const mlEphantManagerMachine2 = setup({
       }
     },
     cacheSetup: assign({
+      conversationId: ({ event }) => {
+        assertEvent(event, MlEphantManagerTransitions2.CacheSetupAndConnect)
+
+        if (event.conversationId) {
+          return event.conversationId
+        }
+
+        return undefined
+      },
```

Comment:
Yeah makes sense! I actually wanted to go for something like that at first but never gave it a proper shot.